packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.component.tsx

```diff
@@ -0,0 +1,216 @@
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {rows.map((row) => (
+                  <React.Fragment key={row.id}>
+                    <TableExpandRow {...getRowProps({ row })}>
+                      {row.cells.map((cell) => (
+                        <TableCell key={cell.id}>{cell.value}</TableCell>
+                      ))}
+                    </TableExpandRow>
+                    {row.isExpanded && (
+                      <TableExpandedRow className={styles.expandedRow} colSpan={headers.length + 2}>
+                        <div className={styles.EncounterObservations}>
+                          <EncounterObservations encounterUuid={row.id} />
+                        </div>
+                      </TableExpandedRow>
+                    )}
+                  </React.Fragment>
+                ))}
+              </TableBody>
+            </Table>
+            {encounters.length === 0 && <p className={styles.dataTableRow}>No encounters found.</p>}
+          </TableContainer>
+        );
+      }}
+    </DataTable>
+  );
+};
+
+interface SingleVisitDetailComponentProps {
+  visit: Visit;
+}
+
+const SingleVisitDetailComponent: React.FC<SingleVisitDetailComponentProps> = ({ visit }) => {
+  const { t } = useTranslation();
+  const [listView, setView] = useState<boolean>(true);
+  const encounters = useMemo(
+    () =>
+      visit.encounters.map((encounter: OpenmrsResource, ind) => ({
+        id: encounter.uuid,
+        time: dayjs(encounter.encounterDateTime).format('hh:mm'),
+        encounterType: encounter.encounterType.display,
+        provider: encounter.encounterProviders.length > 0 ? 'Provider' : '',
+      })),
+    [visit],
+  );
+
+  return (
+    <div className={styles.visitsDetailWidgetContainer}>
+      <div className={styles.visitsDetailHeaderContainer}>
+        <h4 className={styles.productiveHeading02}>
+          {visit.visitType.display}
+          <br />
+          <p className={styles.bodyLong01}>{formatDateTime(visit.startDatetime)}</p>
+        </h4>
+        <div className={styles.toggleButtons}>
+          <Button
+            className={`${styles.toggle} ${listView ? styles.toggleActive : ''}`}
+            size="small"
+            kind="ghost"
+            onClick={() => setView(true)}>
+            {t('All Encounters', 'All Encounters')}
+          </Button>
+          <Button
+            className={`${styles.toggle} ${!listView ? styles.toggleActive : ''}`}
+            size="small"
+            kind="ghost"
+            onClick={() => setView(false)}>
+            {t('Visit Summary', 'Visit Summary')}
+          </Button>
+        </div>
+      </div>
+      {listView && visit?.encounters && <EncounterListDataTable encounters={encounters} />}
+    </div>
+  );
+};
+
+interface VisitOverviewComponentProps {
+  patientUuid: string;
+}
+
+function VisitDetailOverviewComponent({ patientUuid }: VisitOverviewComponentProps) {
+  let A = [1, 2, 3, 4, 5];
```

Comment:
What's this?

Reply:
Ohh sorry, I kept it before making the component structure. Didn't clean the code properly. Will take care of it from now on.
Thanks!